src/mmrelay/meshtastic/connection.py

```
@@ -0,0 +1,1364 @@
+                                with facade._ble_executor_lock:
+                                    if (
+                                        facade._ble_future
+                                        and not facade._ble_future.done()
+                                    ):
+                                        facade.logger.debug(
+                                            "BLE worker busy; skipping interface creation for %s",
+                                            ble_address,
+                                        )
+                                        raise TimeoutError(
+                                            f"BLE interface creation already in progress for {ble_address}."
+                                        )
```

Ruff (0.15.9)

[warning] Abstract `raise` to an inner function

(TRY301)